README.md

```diff
@@ -1,2 +1,199 @@
-# Fluent
-A brand new framework to build amazing android applications
+
+# Fluent 
+
+Fluent is a lightweight framework thats helps you to create an entirely Android Application following SOLID pattern with an unidirectional flow reactive or not.
+
+The main concepts in this architecture are : `State`, `View`, `Store`, `Job` and `Hub` 
+
+---
+>![fluent diagram](https://image.ibb.co/cAXPhH/Screen_Shot_2018_04_11_at_12_09_41.png)
+---
+
+## State
+
+We recommend to define one `State` for each one of your screens.
+
+It holds just `type() : StateType` and should be inherit to fit your needs. As the example above:
+
+```kotlin
+data class LoginState(val type: StateType = StateType.Initial) : State {
+  override fun type() = type
+  fun setType(newType: StateType) = copy(type = newType)
+}
+```
+
+### StateType
+
+We already have some default kind of states, you can create your own type inheriting this class.
+The default states are: `Initial`, `Loading`, `Success` and `Error`
+
+---
+
+>For now ahed we can have two kind of approaches with this framework: A non reactive and a total reactive way.
+
+## Non reactive way
+
+### View
+
+### Store
+
+### Job
+
+### Hub
+
+
+## Reactive way
+
+We work with Reactive Extension to follow the Reactive Manifesto
+
+### View
+
+That's the reference to your screen itself, you should create your kind of `View` that has specific `State` attached. 
+
+```kotlin
+interface LoginView : View<LoginState> {
+  fun doLoginClicks(): Observable<Unit>
+  fun activityResults(): Observable<Pair<Int, Intent>>
+}
+```
+
+> Stay aware to think in a way to declare all the user's actions and screen's actions related as `Observable`.
+
+Your `Activity` or `Fragment` needs to implement `View`
+
+```kotlin
+class LoginActivity : AppCompatActivity(), LoginView {
+    override fun bind(newState: LoginState) { ... }
+}
+```
+
+### RxStore
+
+It receives and manages all the possible states of a `View` so you don't need to worry, because we take care about statefulness and performance.
+
+To create one you just need to pass an `State` class as parameter and define the related initial state:
+
+```kotlin
+RxStore(LoginState(StateType.Initial))
+```
+
+> `RxStore` is an implementation of `Store` that works with Reactive Extensions
+
+The stream created by it can be handled inside the `View` implementation.
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_login)
+
+    store.stateChanges()
+         .observeOn(AndroidSchedulers.mainThread())
+         .subscribe { bind(it) }
+}
+
+override fun bind(newState: LoginState) {
+   when (newState.type) {
+     StateType.Success -> success()
+     StateType.Error -> error()
+     StateType.Loading -> loading()
+   }
+}
+```
+
+
+### RxJob
+
+Small units of work separated from the other layers and that warns the `RxStore` layer when the job is finished successfully or not.
```

Comment:
This first paragraph should also be moved to the first section. We can also improve it. A job is a little piece of work which might look like an atomic operation. It's the only place where you can generate new states and push them to the store.